### ISSUE TITLE
strange descriptor loss

### ISSUE THREAD

**Issue by XhqGlorry11** [rpautrat/SuperPoint#277]
@rpautrat Congrats to your nice work! I'm trying to reproduce training superpoint-coco but run into strange descriptor loss. I set: lambda_d  = 800, lambda_loss = 1 and other settings same with [https://github.com/rpautrat/SuperPoint/blob/master/superpoint/configs/superpoint_coco.yaml](url)
My positive_dist starts around 1.8 and negative_dist starts around 1e-6. But in your training log, positive_dist starts around ~2 and negative_dist starts around 0.4~0.5.
Can you help me check what may be wrong? Thx~~

**Comment by rpautrat** [comment]
Hi,

I trained superpoint-coco from scratch, so the initial values of positive_dist and negative_dist are not really meaningful: it depends on the randomization of the initialization weights.

What matters are the final values, and I think that your values are quite close to my training. I get on average positive_dist = 0.01 and negative_dist = 0.04.

**Comment by XhqGlorry11** [comment]
@rpautrat Thank you for your reply. It seems that you set lambda_d=0.05, lambda_loss = 10000 in superpoint_coco.yml[https://github.com/rpautrat/SuperPoint/blob/master/superpoint/configs/superpoint_coco.yaml](url), which is quite different with paper's setting. What's the main reason?

**Comment by rpautrat** [comment]
The values from the official paper did not work for me. I had to tune these parameters, and the lambda_d = 800, lambda_loss = 1 were the best parameters I found for our released superpoint model.

Since then, I added a normalization in the descriptor loss (https://github.com/rpautrat/SuperPoint/commit/95d1cfd2c2a7a06169ceb63cde01d4d4e3b72cb5) and adapted the values to lambda_d=0.05, lambda_loss = 10000, again by tuning the parameters. So I guess these values should be the right ones to use.

**Comment by XhqGlorry11** [comment]
@rpautrat I notice that you are using descriptors_raw, which is descriptors BEFORE resize_bilinear, to calculate descriptor loss between image&warped_image, but using descriptors, which is descriptors AFTER resize_bilinear, to match. While in the paper, loss is calculated using descriptors AFTER resize_bilinear. Do you observe better result in your implementation way?

**Comment by GuoBo98** [comment]
> The values from the official paper did not work for me. I had to tune these parameters, and the lambda_d = 800, lambda_loss = 1 were the best parameters I found for our released superpoint model.
> 
> Since then, I added a normalization in the descriptor loss ([95d1cfd](https://github.com/rpautrat/SuperPoint/commit/95d1cfd2c2a7a06169ceb63cde01d4d4e3b72cb5)) and adapted the values to lambda_d=0.05, lambda_loss = 10000, again by tuning the parameters. So I guess these values should be the right ones to use. Feel free to tune them more of course!

Hello, may I ask why you added a normalization in the descriptor loss ?

**Comment by rpautrat** [comment]
There was a short explanation for this here: https://github.com/rpautrat/SuperPoint/issues/95#issuecomment-559377632
